internal/route/node_filter.go

```diff
@@ -118,66 +117,57 @@ func (f *IsCloseToGlobalMaxHeight) Apply(
 	_ metadata.RequestMetadata,
 	upstreamConfig *config.UpstreamConfig,
 ) bool {
-	upstreamStatus := f.healthCheckManager.GetUpstreamStatus(upstreamConfig.ID)
-	check := upstreamStatus.BlockHeightCheck
+	status := f.chainMetadataStore.GetBlockHeightStatus(upstreamConfig.GroupID, upstreamConfig.ID)
 
-	checkIsHealthy := check.GetError() == nil
-	if !checkIsHealthy {
+	if status.Error != nil {
 		f.logger.Debug("IsCloseToGlobalMaxHeight failed: most recent health check did not succeed.",
-			zap.Error(check.GetError()),
+			zap.Error(status.Error),
 			zap.String("UpstreamID", upstreamConfig.ID),
 		)
 
 		return false
 	}
 
-	maxHeight := f.chainMetadataStore.GetGlobalMaxHeight()
-	upstreamHeight := check.GetBlockHeight()
-	isClose := upstreamHeight+f.maxBlocksBehind >= maxHeight
-
-	if isClose {
+	if status.BlockHeight+f.maxBlocksBehind >= status.GlobalMaxBlockHeight {
```

Comment:
nit: should we leave the `isClose` variable here or add a comment on this? Mainly thinking about readability. The function name is `IsClose...` so I think it help with tying together the key line(s) of code w/ the function's main purpose.